lib/file-finder.js

```diff
@@ -113,28 +115,31 @@ class FileFinder {
                         // Add the file path to the cache of each directory searched.
                         for (let j = 0; j < searched; j++) {
                             cache[dirs[j]].push(filePath);
+                            yield filePath;
```

Comment:
I think `yield filePath` should be used after this `for` loop, right before the following `break` statement. As currently written, this will yield the same file path multiple times, rather than yielding each file path from a directory once.

(I think this will also fix the issue in https://github.com/eslint/eslint/pull/8638#discussion_r118831995.)